rita/src/rita_client/exit_manager/mod.rs

```diff
@@ -1,14 +1,91 @@
 use actix::prelude::*;
 
-use althea_types::Identity;
+use reqwest;
 
-use settings::RitaClientSettings;
+use althea_types::{ExitClientIdentity, Identity};
+
+use settings::{RitaClientSettings, RitaCommonSettings};
 use SETTING;
 
 use rita_client::rita_loop::Tick;
 use rita_client::traffic_watcher::{TrafficWatcher, Watch};
 
+use althea_types::{ExitServerReply, ExitState};
 use failure::Error;
+use settings::ExitClientDetails;
+use std::net::SocketAddr;
+use KI;
+
+fn linux_setup_exit_tunnel() -> Result<(), Error> {
+    let exit_client = SETTING.get_exit_client();
```

Comment:
so current exit will always be in the registered state?

Reply:
or do we store a state but still go through the registration process on every exit tunnel setup? To be replaced with a login process at some point.

Reply:
we only run setup_exit_tunnel when we know the current exit is in a registered state, and if it is already registered, we don't try registering again

Reply:
then where do we store various connection details? I'm not seeing key and port stored in our config for the long term.

Reply:
here's the relevant part of the settings struct where we store these
```
{
   "exit_client":{
      "current_exit":"exit_a",
      "exits":{
         "exit_a":{
            "general_details":{
               "description":"just a normal althea exit",
               "exit_price":50,
               "netmask":24,
               "server_internal_ip":"172.168.1.254",
               "wg_exit_port":59999
            },
            "id":{
               "eth_address":"0x0101010101010101010101010101010101010101",
               "mesh_ip":"fd00::5",
               "wg_public_key":"vUryCFcZwqkhXtaTnYczhWX5so9DZnu5b0Gx6QvcA34="
            },
            "message":"Registration OK",
            "our_details":{
               "client_internal_ip":"172.168.1.102"
            },
            "registration_port":4875,
            "state":"Registered"
         }
      },
      "reg_details":{
         "email":"1234@gmail.com",
         "zip_code":"1234"
      },
      "wg_listen_port":59999
   }
}
```